crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -863,11 +864,11 @@ impl OlmMachine {
     /// [`decrypt_room_event`]: #method.decrypt_room_event
     pub async fn receive_sync_changes(
         &self,
-        to_device_events: ToDevice,
+        to_device_events: Vec<Raw<AnyToDeviceEvent>>,
```

Comment:
It seems this line change is breaking `crypto-nodejs` and `crypto-js`. In those bindings, `OlmMachine.receiveSyncChanges` receives `to_device_events` as a JSON-encoded string. And that's where this change breaks the code:

* A [`ToDevice`](https://docs.rs/ruma/latest/ruma/api/client/sync/sync_events/v3/struct.ToDevice.html) maps to `{events: [event_i…]}` in JSON,
* A `Vec<Raw<AnyToDeviceEvent>>` maps to `[event_i…]` in JSON.

So the code continues to compile, but the code in the tests are broken because we pass a `{}` by default, instead of a `[]`.

One way to fix this is to keep `to_device_events: ToDevice` in `crypto-js` and `crypto-nodejs`, and to pass `to_device_events.events` when calling into `crypto` (i.e. `matrix_sdk_crypto::OlmMachine::receive_sync_changes(to_device_events.events, …)`.

Reply:
Fixes by https://github.com/gnunicorn/matrix-rust-sdk/pull/1.